src/ophys_etl/modules/video/utils.py

```diff
@@ -56,9 +55,10 @@ def create_downsampled_video(
         Frame rate of the output movie in Hz (set lower than input_hz
         if you want to apply downsampling to the movie)
 
-    kernel_size: Optional[int]
-        Size of the median filter kernel to be applied to the downsampled
-        movie (if None, no median filter will be applied)
+    spatial_filter: Optional[Callable[[np.ndarray], np.ndarray]]
```

Comment:
I've factored out the bulk of the work into `_downsampled_video_array_from_h5` (thanks @morriscb ). Hopefully that makes things a little more streamlined.

I am hesitant to implement these video manipulators as classes because I do not think we want to encourage the practice of carrying the videos around in memory for very long (recall that there will one day be 100 GB videos to process).

That being said (I think I'm repeating myself): we probably need to have a team discussion about how class-based we want to be. Past members were less class-enthusiastic than the current teams seems to be. I'm not sure how we ought to go about making that decision.

Reply:
My thinking is this -- if what you are modeling can be thought of as a _thing_, then it should use a class-based design. Video is a thing, therefore it should use a class-based design. I don't think we necessarily need to carry around the whole video in memory with this approach.

Reply:
I think you've sold me.

After I merge this ticket, I'll open another ticket to create a `Video` class that has `apply_spatial_filter` and `apply_temporal_filter` methods. It will be non-trivial, I think, since it will have to be smart enough to decide based on the size of the movie whether to hold the whole movie in memory or to use a file-backed system and process the movie in chunks.

Anyway: it will be good to have in the future. Thanks for pushing back on this.